midas-contracts/contracts/RedemptionVaultWithBUIDL.sol

```diff
@@ -170,7 +184,15 @@ contract RedemptionVaultWIthBUIDL is RedemptionVault {
         if (contractBalanceTokenOut >= amountTokenOut) return;
 
         uint256 buidlToRedeem = amountTokenOut - contractBalanceTokenOut;
-
+        if (buidlToRedeem < minBuidlToRedeem) {
+            buidlToRedeem = minBuidlToRedeem;
+        }
+        IERC20 buidl = IERC20(buidlRedemption.asset());
+        uint256 buidlBalance = buidl.balanceOf(address(this));
+        require(buidlBalance >= buidlToRedeem, "RVB: buidlToRedeem > balance");
+        if (buidlBalance - buidlToRedeem <= minBuidlToRedeem) {
```

Comment:
Please note that BUIDL has some built-in constraints (but it's not set at the moment) that after if A transfers BUIDL to B, either A would need to transfer all his BUIDL balance, or would require a remaining BUIDL balance >= some threshold.

The threshold can either be `getMinimumHoldingsPerInvestor` or `minUSTokens` in the BUIDL contract. See BUIDL token for details: https://etherscan.deth.net/token/0x7712c34205737192402172409a8f7ccef8aa2aec#code.

So it is recommended to create another variable `minBuidlBalance`, and make the check here:
```solidity
if (buidlBalance - buidlToRedeem <= minBuidlToRedeem || buidlBalance - buidlToRedeem <= minBuidlBalance) {
  buidlToRedeem = buidlBalance;
}
```

Note: https://github.com/sherlock-audit/2024-08-midas-minter-redeemer-judging/issues/128 issue talks about this issue.

Reply:
https://github.com/RedDuck-Software/midas-contracts/pull/74

Reply:
> [RedDuck-Software/midas-contracts#74](https://github.com/RedDuck-Software/midas-contracts/pull/74)

Added a comment in that PR

Reply:
resolved (fixed in https://github.com/RedDuck-Software/midas-contracts/pull/74/commits/8166db788cb0fd8f254ec1958caa7082dc9e7932)

Reply:
LGTM

> resolved (fixed in [RedDuck-Software/midas-contracts@8166db7](https://github.com/RedDuck-Software/midas-contracts/commit/8166db788cb0fd8f254ec1958caa7082dc9e7932))